tests/src/JIT/HardwareIntrinsics/X86/General/VectorHelpers.cs

```diff
@@ -0,0 +1,253 @@
+        }
+        else if (typeof(T) == typeof(ulong))
+        {
+            return Avx.StaticCast<ulong, T>(Avx2.Add(Avx.StaticCast<T, ulong>(left), Avx.StaticCast<T, ulong>(right)));
+        }
+        else if (typeof(T) == typeof(float))
+        {
+            return Avx.StaticCast<float, T>(Avx.Add(Avx.StaticCast<T, float>(left), Avx.StaticCast<T, float>(right)));
+        }
+        else if (typeof(T) == typeof(double))
+        {
+            return Avx.StaticCast<double, T>(Avx.Add(Avx.StaticCast<T, double>(left), Avx.StaticCast<T, double>(right)));
+        }
+        else
+        {
+            throw new NotSupportedException();
+        }
+    }
+
+    public static Vector128<T> SetAllVector128<T>(T value) where T : struct
+    {
+        if (typeof(T) == typeof(byte))
+        {
+            return Sse.StaticCast<byte, T>(Sse2.SetAllVector128(Convert.ToByte(value)));
+        }
+        else if (typeof(T) == typeof(sbyte))
+        {
+            return Sse.StaticCast<sbyte, T>(Sse2.SetAllVector128(Convert.ToSByte(value)));
+        }
+        else if (typeof(T) == typeof(short))
+        {
+            return Sse.StaticCast<short, T>(Sse2.SetAllVector128(Convert.ToInt16(value)));
+        }
+        else if (typeof(T) == typeof(ushort))
+        {
+            return Sse.StaticCast<ushort, T>(Sse2.SetAllVector128(Convert.ToUInt16(value)));
+        }
+        else if (typeof(T) == typeof(int))
+        {
+            return Sse.StaticCast<int, T>(Sse2.SetAllVector128(Convert.ToInt32(value)));
+        }
+        else if (typeof(T) == typeof(uint))
+        {
+            return Sse.StaticCast<uint, T>(Sse2.SetAllVector128(Convert.ToUInt32(value)));
+        }
+        else if (typeof(T) == typeof(long))
+        {
+            return Sse.StaticCast<long, T>(Sse2.SetAllVector128(Convert.ToInt64(value)));
+        }
+        else if (typeof(T) == typeof(ulong))
+        {
+            return Sse.StaticCast<ulong, T>(Sse2.SetAllVector128(Convert.ToUInt64(value)));
+        }
+        else if (typeof(T) == typeof(float))
+        {
+            return Sse.StaticCast<float, T>(Sse.SetAllVector128(Convert.ToSingle(value)));
+        }
+        else if (typeof(T) == typeof(double))
+        {
+            return Sse.StaticCast<double, T>(Sse2.SetAllVector128(Convert.ToDouble(value)));
+        }
+        else
+        {
+            throw new NotSupportedException();
+        }
+    }
+
+    // This helper function is a workaround for the bug of Sse2.SetZeroVector128<float>
+    // https://github.com/dotnet/coreclr/pull/17691
+    public static Vector128<T> SetZeroVector128<T>() where T : struct
+    {
+        if (typeof(T) == typeof(float))
+        {
+            return Sse.StaticCast<float, T>(Sse.SetZeroVector128());
+        }
+        else
+        {
+            return Sse2.SetZeroVector128<T>();
+        }
+    }
```

Comment:
@CarolEidt @tannergooding Without the fix of https://github.com/dotnet/coreclr/pull/17691, users have to write something like this as the workaround. However, it does not work because
1. RyuJIT checks the type arguments in the importer, so that the above code cannot be compiled on `float` with Debug build
```
Assert failure(PID 15536 [0x00003cb0], Thread: 17436 [0x441c]): Assertion failed 'insOfHWIntrinsic(intrinsic, baseType) != INS_invalid' in 'IntelHardwareIntrinsicTest:SetZeroVector128():struct' (IL size 55)

    File: d:\workspace\coreclr\src\jit\hwintrinsicxarch.cpp Line: 816
    Image: D:\workspace\coreclr\bin\tests\Windows_NT.x64.Debug\Tests\Core_Root\CoreRun.exe
```
2. the current SSE2 `SetZeroVector128` has no `HW_Flag_OneTypeGeneric` flag so that cannot throw the exception for non-numeric type arguments, which is a critical inconsistent behavior.

https://github.com/dotnet/coreclr/pull/17691 will fix these two issue, please re-consider that PR.

Reply:
@CarolEidt ping?

Reply:
Hmm... OK, will do.